OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -127,6 +127,9 @@ public virtual void DoAttack(Actor self, Target target, IEnumerable<Armament> ar
 			if (armaments == null && !CanAttack(self, target))
```

Comment:
Shouldn't this be `armaments == null || !CanAttack(self, target)`? (or, not and).  `CanAttack` includes a check for `IsTraitPaused` already.